open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/utils/ConsentRetrievalUtil.java

```diff
@@ -256,4 +272,429 @@ public static JSONArray appendDummyAccountID() {
         return accountsJSON;
 
     }
+
+    /**
+     * Method that invokes the relevant methods to populate data for each flow.
+     *
+     * @param consentResource Consent Resource parameter containing consent related information retrieved from database
+     * @return ConsentDataJson array
+     */
+    public static JSONArray getConsentData(ConsentResource consentResource) throws ParseException {
+
+        JSONArray consentDataJSON = getConsentRetrievalDataSet(consentResource);
+        String receiptString = consentResource.getReceipt();
+        Object receiptJSON = new JSONParser(JSONParser.MODE_PERMISSIVE).parse(receiptString);
+
+        JSONObject receipt = (JSONObject) receiptJSON;
+        JSONObject data = (JSONObject) receipt.get(ConsentExtensionConstants.DATA);
+        JSONObject controlParameters = (JSONObject) data.get(ConsentExtensionConstants.CONTROL_PARAMETERS);
```

Comment:
Similar to https://github.com/wso2/financial-open-banking/pull/38/files#r1492185923

Reply:
Fixed in [a6d2b0b]